routers/api/v1/misc/nodeinfo.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package misc
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/structs"
+)
+
+// NodeInfo returns the NodeInfo for the Gitea instance to allow for federation
+func NodeInfo(ctx *context.APIContext) {
+	// swagger:operation GET /nodeinfo miscellaneous getNodeInfo
+	// ---
+	// summary: Returns the nodeinfo of the Gitea application
+	// produces:
+	// - application/json
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/NodeInfo"
+
+	nodeInfo := &structs.NodeInfo{
+		Version: "2.1",
+		Software: structs.NodeInfoSoftware{
+			Name:       "gitea",
```

Comment:
Is the name intended to be the general-purpose name (`gitea`),
the hostname (i.e. `try.gitea.io`), or the instance-specific name (i.e. `General Purpose Gitea`)?
If it is the first one, which I assume, will the instance-specific name and hostname be sent somewhere else as well, or will it simply be omitted?

Reply:
See https://github.com/jhass/nodeinfo/blob/main/schemas/2.1/schema.json

It's "The canonical name of this server software."